aux_process.c

```diff
@@ -0,0 +1,1503 @@
+    printf("    [-t --tick_len]: integer value indicating the tick length in tenths of a second (default is 4)\n");
+    printf("    [-v --verbose]: write log entries to stdout\n");
+    printf("\n");
+}
+
+/*-------------------------------------------------------------------------
+ * Function: parse_command_line
+ *
+ * Purpose:  Parse the options that a user specifies
+ *
+ * Return:   Success:    0
+ *
+ *           Failure:    -1
+ *-------------------------------------------------------------------------
+ */
+static int
+parse_command_line(int argc, char *argv[], handler_t *hand)
+{
+    int opt;
+    aux_long_options long_options[] =
+    {
+        {"vfd_config=", require_arg, 'c'},
+        {"help", no_arg, 'h'},
+        {"log_file_path=", require_arg, 'l'},
+        {"md_file_chksum=", require_arg, 'm'},
+        {"polls_per_tick=", require_arg, 'p'},
+        {"stats=", require_arg, 's'},
+        {"tick_len=", require_arg, 't'},
+        {"verbose=", require_arg, 'v'},
+        {NULL, 0, 0}
+    };
+
+    /* Initialize the command line options */
+    hand->log_file_path = NULL;
+    hand->log_file = NULL;
+    hand->output = NULL;
+    hand->polls_per_tick = 10;
+    hand->print_stats = false;
+    hand->tick_len = 4;
+    hand->verbose = false;
+    hand->vfd_config = NULL;
+    hand->updater_path = NULL;
+    hand->md_file_path = NULL;
+    hand->md_chksum_path = NULL;
+    hand->md_file = NULL;
+    hand->md_chksum_file = NULL;
+    hand->num_mdfile_checksums = 0;
+    hand->md_file_checksums = NULL;
+
+    if (argc >= 2) {
+        hand->updater_path = strdup(argv[argc-1]);
+        hand->md_file_path = strdup(argv[argc-2]);
+        argc -= 2;
+    } else {
+        fprintf(stderr, "command-line options must have the path for the metadata file and the updater files\n");
+        goto error;
+    }
+
+    /* 
+     * aux_get_options supports both POSIX and Windows 
+     */
+    while((opt = aux_get_options(argc, argv, "c:hl:m:p:st:v", long_options)) != EOF)
+    {
+        switch(opt)
+        {
+            case 'c':
+                /* The configuration string for the VFD stack */
+                if(aux_optarg) {
+                    fprintf(stdout, "The configuration string for the VFD stack:\t%s\n", aux_optarg);
+                    hand->log_file_path = strdup(aux_optarg);
+                } else
+                    fprintf(stderr, "aux_optarg is null\n");
+                break;
+            case 'h': 
+                fprintf(stdout, "Help page:\n");
+                usage();
+
+                exit(0);
+
+                break;
+            case 'l':
+                /* The log file */
+                if(aux_optarg) {
+                    fprintf(stdout, "The log file:\t\t\t\t\t\t%s\n", aux_optarg);
+                    hand->log_file_path = strdup(aux_optarg);
+                } else
+                    fprintf(stderr, "aux_optarg is null\n");
+                break;
+            case 'm':
```

Comment:
Add description that this option (plus the associated routines) is added for testing purpose.
The description can be added somewhere you think appropriate.

Reply:
Done.